src/config.rs

```diff
@@ -216,13 +217,98 @@ pub struct AuthorityKeyIdentifierExtension {
     pub issuer: bool,
 }
 
+/// The `CertificatePolicy` enum represents the set of KDL nodes that `pki-playground` can map to
+/// OIDs. Configs may also provide OIDs in their string forms using the `oid` node.
+#[derive(knuffel::Decode, Debug)]
+pub enum CertificatePolicy {
+    /// Initial attestation policy OID from [DICE Certificate
+    /// Profiles](https://trustedcomputinggroup.org/resource/dice-certificate-profiles/) §5.1.5.3
+    TcgDiceKpAttestInit,
+    /// Local attestation policy OID from [DICE Certificate
+    /// Profiles](https://trustedcomputinggroup.org/resource/dice-certificate-profiles/) §5.1.5.4
+    TcgDiceKpAttestLoc,
+    /// Initial assertion policy OID from [DICE Certificate
+    /// Profiles](https://trustedcomputinggroup.org/resource/dice-certificate-profiles/) §5.1.5.5
+    TcgDiceKpAssertInit,
+    /// Local assertion policy OID from [DICE Certificate
+    /// Profiles](https://trustedcomputinggroup.org/resource/dice-certificate-profiles/) §5.1.5.6
+    TcgDiceKpAssertLoc,
+    /// Embedded certificate authority (ECA) policy OID from [DICE Certificate
+    /// Profiles](https://trustedcomputinggroup.org/resource/dice-certificate-profiles/) §5.1.5.7
+    TcgDiceKpEca,
+    /// Initial identity policy OID from [DICE Certificate
+    /// Profiles](https://trustedcomputinggroup.org/resource/dice-certificate-profiles/) §5.1.5.1
+    TcgDiceKpIdentityInit,
+    /// Local identity policy OID from [DICE Certificate
+    /// Profiles](https://trustedcomputinggroup.org/resource/dice-certificate-profiles/) §5.1.5.2
+    TcgDiceKpIdentityLoc,
+    /// Platform identity policy from [OANA x.509 certificate policy
+    /// terms](https://github.com/oxidecomputer/oana#asn1-object-identifiers)
+    OanaPlatformIdentity,
+    /// RoT code signing development policy from [OANA x.509 certificate policy
+    /// terms](https://github.com/oxidecomputer/oana#asn1-object-identifiers)
+    OanaRotCodeSigningDevelopment,
+    /// RoT code signing release policy from [OANA x.509 certificate policy
+    /// terms](https://github.com/oxidecomputer/oana#asn1-object-identifiers)
+    OanaRotCodeSigningRelease,
+    /// `oid` node taking an OID string argument
+    Oid(#[knuffel(argument)] String),
+}
+
+impl TryFrom<&CertificatePolicy> for PolicyInformation {
+    type Error = miette::Error;
+
+    /// Map `CertificatePolicy` variants to the appropriate `PolicyInformation` struct. This is
+    /// required as part of our conversion from the KDL to the DER certificate encoding.
+    fn try_from(value: &CertificatePolicy) -> Result<Self> {
```

Comment:
Is it possible we could use something like [const_oid::db::Database::by_name](https://docs.rs/const-oid/latest/const_oid/db/struct.Database.html#method.by_name) to map from strings to OIDs? I worry that this hand-coded translation will be brittle and hard to update.

Reply:
Agree there's lots of room for improvement: #69